maintenance/pre-commit-check.sh

```diff
@@ -0,0 +1,8 @@
+#! /usr/bin/env bash
+
+set -e
```

Comment:
```suggestion
#!/usr/bin/env bash
set -Eeuo pipefail
```

No space in shebang ; add some more error handling options.